common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java

```diff
@@ -105,6 +106,19 @@ static String urlEncode(String str) {
         }
     }
 
+    /**
+     * Redact any `user:password` that is part of the URL.
+     */
+    static String redactUserInfo(String url) {
+        try {
+            URI uri = new URI(url);
+            String userInfoPart = uri.getUserInfo() + '@';
+            return url.replace(userInfoPart, "******@");
```

Comment:
This does not seem to be null-safe (from the getUserInfo Javadoc). (for Gradle and Maven)

Reply:
Wow, this was [totally handled in the original PR](https://github.com/gradle/gradle-enterprise-build-config-samples/pull/160/files#diff-52523462d2092ee6f6abe1aa4ada47ac1fd2195f039cd20574562c2688c51b5fR109-R111) and I missed it when copy-pasting into my solution. 🤦 .

Reply:
Fixed